ui/pages/permissions-connect/connect-page/connect-page.tsx

```diff
@@ -98,103 +110,157 @@ export const ConnectPage: React.FC<ConnectPageProps> = ({
   const t = useI18nContext();
   const trackEvent = useContext(MetaMetricsContext);
 
+  // const shouldPromptToCreateSolanaAccount =
+  //   request.metadata?.promptToCreateSolanaAccount;
+
   const requestedCaip25CaveatValue = getRequestedCaip25CaveatValue(
     request.permissions,
   );
-  const requestedAccounts = getEthAccounts(requestedCaip25CaveatValue);
-  const requestedChainIds = getPermittedEthChainIds(requestedCaip25CaveatValue);
 
-  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
-  const [nonTestNetworks, testNetworks] = useMemo(
+  const requestedCaipAccountIds = getAllAccountsFromCaip25CaveatValue(
```

Comment:
(For later)

This is a meaty component 😭 Between this and the `review-permissions-page` component there's a lot of logic and data transformation happening at the UI layer. It feels like there ought to be some other place we can put this. Not quite sure but lifting it out of the component to another file in the same directory might be a good first step. I guess you've already got some functions in `utils.ts`, perhaps we can extract more.

Reply:
Alex has started on this here https://github.com/MetaMask/metamask-extension/pull/31641